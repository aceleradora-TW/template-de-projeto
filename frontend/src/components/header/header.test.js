import React from 'react';
import { render } from '@testing-library/react';
import Header from './';

describe('Header', () => {
  test('Should renders component when there is title', () => {
    const headerProps = {
      title: 'fake-title'
    };
    const { container, getByText } = render(<Header {...headerProps} />);
    expect(getByText(headerProps.title)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('Should renders component with warning when there is no title', () => {
    const consoleError = jest.spyOn(console, 'error');
    const headerProps = {};
    const { container } = render(<Header {...headerProps} />);
    expect(consoleError).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });


});
