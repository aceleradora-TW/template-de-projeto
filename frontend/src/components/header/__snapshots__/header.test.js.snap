// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header has className 1`] = `
<header
  class="fake-class"
>
  fake-title
  <a
    class="App-link"
    href="https://reactjs.org"
    rel="noopener noreferrer"
    target="_blank"
  >
    Learn React
  </a>
</header>
`;
